Task aop

Sindicat007 opened the pull request.

Sindicat007 committed:
Задачи по AOP

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

Sindicat007 committed:
Задачи по AOP

Signed-off-by: Sindicat007 <sindicat007@ya.ru>

RinatBeybutov commented on AOP/src/main/java/ru/maxima/aop/aspect/TimeOutAspect.java:
```diff
@@ -0,0 +1,28 @@
+@Aspect
+public class TimeOutAspect {
+
+    private static long time;
```
а каким значением будет эта переменная инициализироваться? 
да и переименовать лучше в lastExecutionTime

RinatBeybutov commented on AOP/src/main/java/ru/maxima/aop/aspect/TimeOutAspect.java:
```diff
@@ -0,0 +1,28 @@
+    @Around("execution(* ru.maxima.aop.service.MyService.doWithTimeOut(..))")
+    public Object timeoutExecution(ProceedingJoinPoint joinPoint) throws Throwable {
+        long startTime = System.currentTimeMillis();
+        if (time == 0) {
```
может сделать здесь выход из метода, если меньше 5 сек прошло? так как-то проще выглядит реализация, чем ждать и потом пропускать

Sindicat007 committed:
Добавил инициализацию меременной lastExecutionTime,
Заменил задержку потока на выход из метода, настроил правильный подсчет времени

Signed-off-by: Sindicat007 <sindicat007@ya.ru>
